plugins/generic/usageStats/UsageStatsPlugin.inc.php

```diff
@@ -321,6 +380,31 @@ function _writeUsageEventInLogFile($usageEvent) {
 		}
 		fclose($fp);
 	}
+
+	//
+	// Private helper methods.
+	//
+	/**
+	* Hash (SHA256) the given IP using the given SALT.
+	*
+	* NB: This implementation was taken from OA-S directly. See
+	* http://sourceforge.net/p/openaccessstati/code-0/3/tree/trunk/logfile-parser/lib/logutils.php
+	* We just do not implement the PHP4 part as OJS dropped PHP4 support.
```

Comment:
Just to be safe, did we dropped PHP4 support for OJS 2.x lines? I am not sure about that, might be worth it to ask @asmecher